contracts/gateway/SimpleStake.sol

```diff
@@ -15,101 +15,133 @@ pragma solidity ^0.5.0;
 // limitations under the License.
 //
 // ----------------------------------------------------------------------------
-// ValueChain: SimpleStake
-// 
+//
 // http://www.simpletoken.org/
 //
 // ----------------------------------------------------------------------------
 
 import "./EIP20Interface.sol";
 import "../lib/SafeMath.sol";
-import "./ProtocolVersioned.sol";
 
 /**
- *  @title SimpleStake contract which implements ProtocolVersioned.
+ *  @title SimpleStake contract
  *
- *  @notice Stakes the value of an EIP20 token on Ethereum
- *          for a utility token on the OpenST platform.
+ *  @notice This holds staked EIP20 tokens for a gateway.
  */
-contract SimpleStake is ProtocolVersioned {
+contract SimpleStake {
     using SafeMath for uint256;
 
+
     /* Events */
 
-    event ReleasedStake(address indexed _protocol, address indexed _to, uint256 _amount);
+    /* Emitted when amount in un-staked */
+    event ReleasedStake(
+        address indexed _gateway,
+        address indexed _to,
+        uint256 _amount
+    );
+
 
     /* Storage */
 
     /** EIP20 token contract that can be staked. */
     EIP20Interface public eip20Token;
 
+    /** EIP20 gateway address. */
     address public gateway;
 
-    /* Public functions */
+
+    /* Modifiers */
+
+    /** checks that only gateway can call a particular function. */
+    modifier onlyGateway() {
+        require(
+            msg.sender == gateway,
+            "Only gateway can call the function."
+        );
+        _;
+    }
+
+    /* Constructor */
 
     /**
      *  @notice Contract constructor.
      *
-     *  @dev Sets the protocol and the EIP20 token to stake.
-     *
      *  @param _eip20Token EIP20 token that will be staked.
-     *  @param _gateway Gateway contract that governs staking.
+     *  @param _gateway EIP20Gateway contract that governs staking.
      */
     constructor(
         EIP20Interface _eip20Token,
         address _gateway
     )
-        ProtocolVersioned(_gateway)
         public
     {
+        require(
+            address(_eip20Token)!= address(0),
+            "Token contract address must not be zero."
+        );
+        require(
+            _gateway != address(0),
+            "Gateway contract address must not be zero."
+        );
+
         eip20Token = _eip20Token;
         gateway = _gateway;
     }
 
+
+    /* External functions */
+
     /**
-     *  @notice Public function releaseTo.
+     *  @notice This allows gateway to release staked amount to provided address.
      *
-     *  @dev Callable only by protocol. Allows the protocol to release the staked amount
-     *       into provided address. The protocol MUST be a contract that sets the rules
-     *       on how the stake can be released and to who.
-     *       The protocol takes the role of an "owner" of the stake.
+     *  @dev Only gateway contract can call this method which defines set of rules
```

Comment:
I think this could be added to `@notice` instead of `@dev` as this is relevant to anyone looking at the API of this contract.